src/System.Net.Http/tests/FunctionalTests/HttpContentTest.cs

```diff
@@ -0,0 +1,631 @@
+        }
+
+        [Fact]
+        public async Task LoadIntoBufferAsync_ThrowObjectDisposedExceptionInOverriddenAsyncMethod_ThrowsHttpRequestException()
+        {
+            var content = new MockContent(new ObjectDisposedException(""), MockOptions.ThrowInAsyncSerializeMethods);
+
+            HttpRequestException ex = await Assert.ThrowsAsync<HttpRequestException>(() => content.LoadIntoBufferAsync());
+            Assert.IsType<ObjectDisposedException>(ex.InnerException);
+        }
+
+        [Fact]
+        public async Task LoadIntoBufferAsync_ThrowIOExceptionInOverriddenAsyncMethod_ThrowsHttpRequestException()
+        {
+            var content = new MockContent(new IOException(), MockOptions.ThrowInAsyncSerializeMethods);
+
+            HttpRequestException ex = await Assert.ThrowsAsync<HttpRequestException>(() => content.LoadIntoBufferAsync());
+            Assert.IsType<IOException>(ex.InnerException);
+        }
+
+        [Fact]
+        public async Task Dispose_GetReadStreamThenDispose_ReadStreamGetsDisposed()
+        {
+            var content = new MockContent();
+            MockMemoryStream s = (MockMemoryStream) await content.ReadAsStreamAsync();
+            Assert.Equal(1, content.CreateContentReadStreamCount);
+
+            Assert.Equal(0, s.DisposeCount);
+            content.Dispose();
+            Assert.Equal(1, s.DisposeCount);
+        }
+
+        [Fact]
+        public void Dispose_DisposeContentThenAccessContentLength_Throw()
+        {
+            var content = new MockContent();
+
+            // This is not really typical usage of the type, but let's make sure we consider also this case: The user
+            // keeps a reference to the Headers property before disposing the content. Then after disposing, the user
+            // accesses the ContentLength property.
+            var headers = content.Headers;
+            content.Dispose();
+            Assert.Throws<ObjectDisposedException>(() => headers.ContentLength.ToString());
+        }
+
+        [Fact]
+        public async Task CopyToAsync_UseStreamWriteByteWithBufferSizeSmallerThanContentSize_ThrowsHttpRequestException()
+        {
+            // MockContent uses stream.WriteByte() rather than stream.Write(): Verify that the max. buffer size
+            // is also checked when using WriteByte().
+            var content = new MockContent(MockOptions.UseWriteByteInCopyTo);
+            await Assert.ThrowsAsync<HttpRequestException>(() => content.LoadIntoBufferAsync(content.GetMockData().Length - 1));
+        }
+        
+        [Fact]
+        public async Task ReadAsStringAsync_EmptyContent_EmptyString()
+        {
+            var content = new MockContent(new byte[0]);
+            string actualContent = await content.ReadAsStringAsync();
+            Assert.Equal(string.Empty, actualContent);
+        }
+
+        [Fact]
+        [ActiveIssue(3343, PlatformID.AnyUnix)]
```

Comment:
This is due to https://github.com/dotnet/corefx/issues/846... master doesn't yet have a properly implemented GetCodePageFromName method, which is returning UTF8 for unrecognized names rather than throwing (https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Globalization/EncodingTable.Unix.cs#L45-L64)